src/main/java/cuchaz/enigma/gui/elements/MenuBar.java

```diff
@@ -274,6 +277,73 @@ public MenuBar(Gui gui) {
 
 			}
 		}
+		{
+			JMenu menu = new JMenu(I18n.translate("menu.collab"));
+			this.add(menu);
+			{
+				JMenuItem item = new JMenuItem(I18n.translate("menu.collab.connect"));
+				menu.add(item);
+				item.addActionListener(event -> {
+					if (this.gui.getController().getClient() != null) {
+						this.gui.getController().disconnectIfConnected(null);
+						return;
+					}
+					ConnectToServerDialog.Result result = ConnectToServerDialog.show(this.gui.getFrame());
+					if (result == null) {
+						return;
+					}
+					this.gui.getController().disconnectIfConnected(null);
+					try {
+						this.gui.getController().createClient(result.getUsername(), result.getIp(), result.getPort());
+					} catch (IOException e) {
+						JOptionPane.showMessageDialog(this.gui.getFrame(), e.toString(), I18n.translate("menu.collab.connect.error"), JOptionPane.ERROR_MESSAGE);
+						this.gui.getController().disconnectIfConnected(null);
+					}
+				});
+				this.connectToServerMenu = item;
+			}
+			{
+				JMenuItem item = new JMenuItem(I18n.translate("menu.collab.server.start"));
+				menu.add(item);
+				item.addActionListener(event -> {
+					if (this.gui.getController().getServer() != null) {
+						this.gui.getController().disconnectIfConnected(null);
+						return;
+					}
+					String result = (String) JOptionPane.showInputDialog(
+							this.gui.getFrame(),
+							I18n.translate("prompt.port"),
+							I18n.translate("prompt.port"),
+							JOptionPane.QUESTION_MESSAGE,
+							null,
+							null,
+							String.valueOf(EnigmaServer.DEFAULT_PORT)
+					);
+					if (result == null) {
+						return;
+					}
+					int port;
+					try {
+						port = Integer.parseInt(result);
+					} catch (NumberFormatException e) {
+						JOptionPane.showMessageDialog(this.gui.getFrame(), I18n.translate("prompt.port.nan"), I18n.translate("prompt.port"), JOptionPane.ERROR_MESSAGE);
+						return;
+					}
+					if (port < 1024 || port >= 65536) {
```

Comment:
Any reason to disallow ports below 1024? I know many operating systems require the first thousand or so ports to be opened with admin permissions, but I don't think we should limit it here